tests/gitgitgadget.test.ts

```diff
@@ -252,20 +252,27 @@ to have included in git.git [https://github.com/git/git].`);
 
     expect(metadata2?.coverLetterMessageId)
         .toMatch(/pull\.1\.v2\.git\.\d+\.gitgitgadget@example\.com/);
-    expect(mails.length).toEqual(5);
+    expect(mails).toHaveLength(5);
     if (await gitCommandExists("range-diff", repo.workDir)) {
         expect(mails[0]).toMatch(/Range-diff vs v1:\n[^]*\n -: .* 4: /);
     }
     expect(await repo.revParse("pr-1/somebody/master-v2")).toBeDefined();
 
     const seriesMeta = await notes.get<IPatchSeriesMetadata>(pullRequestURL);
     expect(seriesMeta).not.toBeNull();
-    expect(seriesMeta!.coverLetterMessageId).not.toBeUndefined();
-    const coverMid: string = seriesMeta!.coverLetterMessageId!;
+    if (!seriesMeta) throw new Error(`assertion failed`); // avoid TS2532
```

Comment:
Not really clear on the use of throwing errors in the tests versus using expect.  Is this just to get past eslint?  I take it you are not a fan of the new `?` (usually replacing `!`) - I have not looked but doc suggested it generates code to check at runtime and I am not a fan of extra code but this is in the test suite.  And that was a run-on sentence.

Reply:
>  Is this just to get past eslint?

It's to get past `tsc`. ESLint is perfectly happy with it, somehow figuring out that `.not.toBeNull()` is plenty enough to ensure what we want to ensure.

>  I take it you are not a fan of the new `?` (usually replacing `!`) - I have not looked but doc suggested it generates code to check at runtime and I am not a fan of extra code but this is in the test suite.

Let me have a quick look at your version. *clicketyclick* ah: https://github.com/webstech/gitgitgadget/blob/41d6a00243add6c935ad9dbe31b5d153cb079c14/tests/gitgitgadget.test.ts#L264

Yes, that looks indeed _a lot_ nicer. Will fix.

Reply:
I even found a way to do this without disabling ESLint: https://github.com/webstech/gitgitgadget/blob/41d6a00243add6c935ad9dbe31b5d153cb079c14/tests/gitgitgadget.test.ts#L269-L270

```js
    const refMid: string | undefined = seriesMeta?.referencesMessageIds?.[0];
```